src/openms/source/CHEMISTRY/SvmTheoreticalSpectrumGenerator.cpp

```diff
@@ -743,6 +743,18 @@ namespace OpenMS
 
     Int simulation_type = (Int)param_.getValue("svm_mode");
 
+    if (add_metainfo)
+    {
+      if (spectrum.getIntegerDataArrays().size() == 0)
+      {
```

Comment:
There are other cases to consider: For example, what happens if the last data array has the same name? Add another one or put values in it?
Right now, I decided for the simplest and less ambiguous solution: just add array if none is in there.
But I can see the point that we might need to come up with a policy or "good practice" guide. Question is how we should proceed here?